### DISCUSSION TITLE
How to do a boosted `hx-redirect`?

### DISCUSSION THREAD

**Discussion by msladecek** [bigskysoftware/htmx#876] [Q&A]
I would like to redirect the user to a different page using the [`hx-redirect` header feature](https://htmx.org/docs/#response-headers).

This triggers a full page reload on the new url.
Is there any way to do it without dropping the current page and loading the next one?
I would like to get behavior like [`hx-boost`](https://htmx.org/attributes/hx-boost/) - the new content is swapped into the page and a new url pushed into the history.

It seems to me that the solution might be to return the same body as the targeted url, and do a `hx-push` in the headers alongside it, but I'm hoping there is some simpler, more transparent solution.

**Comment by gone** [comment]
I've been working on this and we're waiting on CR and merge: https://github.com/bigskysoftware/htmx/pull/832

**Reply by scrhartley** [reply]
For reference, the [`HX-Location`](https://htmx.org/headers/hx-location/) response header has been available since 1.8.0

**Comment by gnat** [comment]
For people who stumble upon this you can also use `hx-target="body" hx-swap="outerHTML" hx-push-url="true"` to do a boosted redirect.

**Reply by johiny** [reply]
yesterday i was searching for hours for this thx man!
